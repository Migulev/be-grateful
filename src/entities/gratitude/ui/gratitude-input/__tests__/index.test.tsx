import { act, fireEvent, render, screen, within } from '@testing-library/react'
import userEvent from '@testing-library/user-event'

import { GratitudeInput } from '..'

vitest.mock('@/shared/libs/context/auth-modal-context', () => ({
  useAuthModal: vitest.fn(() => ({
    setIsOpenAuthModal: vitest.fn(),
  })),
}))

describe('GratitudeInput', () => {
  const onCreateAsyncMock = vitest.fn()

  const mockGratitude = 'Test gratitude'

  it('renders placeholder correctly and prefix "-" ', async () => {
    render(
      <GratitudeInput
        onCreateAsync={onCreateAsyncMock}
        isAuthorized={true}
      />,
    )

    const form = screen.getByRole('form')
    const textarea = screen.getByRole('textbox')

    expect(within(form).getByText('-')).matchSnapshot()
    expect(textarea).toHaveAttribute('placeholder', '...')

    await act(async () => {
      await userEvent.type(textarea, mockGratitude)
    })

    const textareaAfterType = await screen.findByDisplayValue(mockGratitude)
    expect(textareaAfterType).matchSnapshot()
  })

  it('handles Enter key press', async () => {
    render(
      <GratitudeInput
        onCreateAsync={onCreateAsyncMock}
        isAuthorized={true}
      />,
    )

    const form = screen.getByRole('form')
    const textarea = screen.getByRole('textbox')

    await act(async () => {
      await userEvent.type(textarea, mockGratitude)
      fireEvent.keyDown(form, { key: 'Enter' })
    })

    expect(onCreateAsyncMock).toHaveBeenCalledWith(mockGratitude)
  })
})
