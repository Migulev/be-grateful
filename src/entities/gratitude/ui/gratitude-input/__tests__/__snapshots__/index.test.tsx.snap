// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`GratitudeInput > renders placeholder correctly and prefix "-"  1`] = `
<span
  class="absolute ml-8 mt-[9px] text-lg text-muted-foreground"
>
  -
</span>
`;

exports[`GratitudeInput > renders placeholder correctly and prefix "-"  2`] = `
<textarea
  class="flex w-full resize-none rounded border border-input px-3 py-2 placeholder:text-muted-foreground focus-visible:outline-none focus-visible:ring-ring focus-visible:ring-offset-2 disabled:cursor-not-allowed disabled:opacity-50 no-scrollbar overflow-y-auto bg-secondary pl-10 text-lg text-muted-foreground ring-offset-secondary focus-visible:ring-1"
  placeholder="..."
  rows="1"
  style="height: 0px;"
>
  Test gratitude
</textarea>
`;
